src/Form/DataTransformer/ModelToIdPropertyTransformer.php

```diff
@@ -72,9 +80,16 @@ public function __construct(
         $this->toStringCallback = $toStringCallback;
     }
 
+    /**
+     * @param mixed[]|null $value
```

Comment:
Why phpstan/psalm doesn't report that 
```
if (!\is_array($value)) {
    throw new \UnexpectedValueException(sprintf('Value should be array, %s given.', \gettype($value)));
}
```
is useless ?

Reply:
because it could be null

Reply:
There is a `if (empty($value)) {` made earlier

Reply:
Hum, but empty is more like to test if an array is empty.

Reply:
Sure, but what I wanted to said if that `$value` can't be null during the 
```
if (!\is_array($value)) {
```
Check, since the `null` value are already caught by the
```
if (empty($value)) {
```
Check.

Either we should use `@param mixed $value` either we could remove the array check.

Reply:
IMHO we should keep the array check, because phpstan attributes does not enforce valid types.

Reply:
> IMHO we should keep the array check, because phpstan attributes does not enforce valid types.

I'm not against this check. But then, it would be better to use `@param mixed $value`.

If one day we change 
```
if (empty($value)) {
```
to 
```
if (null === $value) {
```
We would end with a phpstan error with our current set up.

Reply:
Changed it back to `mixed`